src/components/utils/debounce.ts

```diff
@@ -0,0 +1,8 @@
+export function debounce(f:(finalQuery: string) => void, delay: number) {
+  let timerId: NodeJS.Timeout;
+
+  return (arg: string) => {
```

Comment:
will be awesome to accept any arguments here for making debounce the universal one
you can use arguments keyword for this purpose

Reply:
have some problems to write rigth types for debounce, hope I successed